### PR TITLE
Use deps

### PR DESCRIPTION
**Pull request by apelisse**
Use github.com/golang/deps to manage dependencies. It's cleaner.

### CODE REVIEW

**Comment by apelisse** [comment]
It's difficult to know what is used and what is not in that repo, since it's doing nothing for the moment. But I removed the code (unused?) that depends on resources from k8s.io/kubernetes for now. I don't think it'd be a good idea to vendor kubernetes in kubectl.

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+
+[[constraint]]
+  branch = "v2"
+  name = "gopkg.in/yaml.v2"
+
+[[constraint]]
+  branch = "master"
```
Comment (mengqiy) [inline Gopkg.toml]: I don't think pin to master branch is good idea.

File: pkg/kinflate/util.go [hunk 2]
```diff
@@ -17,63 +17,17 @@ limitations under the License.
...
 )
 
 const kubeManifestFileName = "Kube-manifest.yaml"
 
-// loadBaseAndOverlayPkg returns:
-// - List of FilenameOptions, each FilenameOptions contains all the files and whether recursive for each base defined in overlay kube-manifest.yaml.
-// - Fileoptions for overlay.
-// - Package object for overlay.
-// - A potential error.
-func loadBaseAndOverlayPkg(f string) ([]resource.FilenameOptions, resource.FilenameOptions, *manifest.Manifest, error) {
```
Comment (mengqiy) [inline pkg/kinflate/util.go]: Don't remove it.

Comment it for now :)

Comment (apelisse) [reply pkg/kinflate/util.go]: Yeah, I'll comment ;-)

File: Gopkg.toml (hunk 1 above)
Comment (apelisse) [reply Gopkg.toml]: This is what `deps`has set by default. what do you suggest?

Comment (mengqiy) [reply Gopkg.toml]: If production kubectl code lives here, I will suggest use `version = "v1.8.3"` to make it more stable.
But no production code lives here now ATM, so I take it back. Using the `master` is fine.
Keep it `master` until kubectl is isolated and moves here.

**Comment by apelisse** [comment]
Tests are failing on this PR since the "verify travis" script doesn't work well with the vendor directory. We need to merge #130 first.

File: pkg/kinflate/util.go (hunk 2 above)
Comment (apelisse) [reply pkg/kinflate/util.go]: Done

**Comment by apelisse** [comment]
@mengqiy I would really appreciate if you could review and maybe approve this change.
@hoegaarden Can you please review this quickly, even if its merged when you have time to do it?

Thank you!

**Comment by mengqiy** [comment]
I have to say using dep to manage `apimachinery` repo will make `kinflate` work harder.

**Comment by mengqiy** [comment]
See: https://github.com/kubernetes/kubectl/pull/144/commits/68011dc6ca9393c263ce924aee714c55b7f646bc#diff-a6c2ab3f6dea554a6f402bf4a9f269cfR73

**Pull request merged**